File: apps/LemonadeNexusMac/Sources/LemonadeNexusMac/Views/NodeDetailView.swift
import SwiftUI
import AppKit

struct NodeDetailView: View {
    let node: TreeNode

    @EnvironmentObject private var appState: AppState
    @State private var isEditing: Bool = false
    @State private var editHostname: String = ""
    @State private var editRegion: String = ""
    @State private var showDeleteConfirmation: Bool = false
    @State private var isSaving: Bool = false
    @State private var statusMessage: String?

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 20) {
                // Header
                headerSection

                Divider()

                // Properties
                propertiesSection

                Divider()

                // Network Info
                networkSection

                Divider()

                // Keys
                keysSection

                // Assignments
                if let assignments = node.assignments, !assignments.isEmpty {
                    Divider()
                    assignmentsSection(assignments)
                }

                Spacer(minLength: 20)

                // Actions
                actionsSection
            }
            .padding(24)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(Color.surfaceDark)
        .onAppear {
            editHostname = node.hostname
            editRegion = node.region ?? ""
        }
        .alert("Delete Node", isPresented: $showDeleteConfirmation) {
            Button("Cancel", role: .cancel) { }
            Button("Delete", role: .destructive) {
                Task { await deleteNode() }
            }
        } message: {
            Text("Are you sure you want to delete \"\(node.hostname)\"? This action cannot be undone.")
        }
    }

    // MARK: - Header

    private var headerSection: some View {
        HStack(spacing: 16) {
            ZStack {
                RoundedRectangle(cornerRadius: 12)
                    .fill(nodeColor.opacity(0.15))
                    .frame(width: 56, height: 56)
                Image(systemName: node.nodeType.sfSymbol)
                    .foregroundColor(nodeColor)
                    .font(.title)
            }

            VStack(alignment: .leading, spacing: 4) {
                if isEditing {
                    TextField("Hostname", text: $editHostname)
                        .font(.title2.bold())
                        .textFieldStyle(.roundedBorder)
                } else {
                    Text(node.hostname)
                        .font(.title2.bold())
                        .foregroundColor(.textPrimary)
                }
                HStack(spacing: 8) {
                    BadgeView(text: node.nodeType.displayName, color: nodeColor)
                    Text(node.id)
                        .font(.caption.monospaced())
                        .foregroundColor(.textTertiary)
                        .lineLimit(1)
                        .truncationMode(.middle)
                }
            }

            Spacer()

            Button(action: { isEditing.toggle() }) {
                Image(systemName: isEditing ? "checkmark.circle.fill" : "pencil.circle")
                    .font(.title2)
                    .foregroundColor(isEditing ? .lemonGreen : .textSecondary)
            }
            .buttonStyle(.plain)
            .help(isEditing ? "Done Editing" : "Edit Node")
        }
    }

    // MARK: - Properties

    private var propertiesSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            SectionHeaderView(title: "Properties", icon: "info.circle")

            VStack(spacing: 10) {
                propertyRow("Node ID", value: node.id, monospaced: true)
                propertyRow("Parent ID", value: node.parent_id, monospaced: true)
                propertyRow("Type", value: node.nodeType.displayName)

                if isEditing {
                    HStack {
                        Text("Hostname")
                            .font(.subheadline)
                            .foregroundColor(.textSecondary)
                            .frame(width: 120, alignment: .leading)
                        TextField("Hostname", text: $editHostname)
                            .textFieldStyle(.roundedBorder)
                    }
                    HStack {
                        Text("Region")
                            .font(.subheadline)
                            .foregroundColor(.textSecondary)
                            .frame(width: 120, alignment: .leading)
                        TextField("Region", text: $editRegion)
                            .textFieldStyle(.roundedBorder)
                    }
                } else {
                    propertyRow("Hostname", value: node.hostname)
                    if let region = node.region {
                        propertyRow("Region", value: region)
                    }
                }
            }
            .cardStyle()
        }
    }

    // MARK: - Network

    private var networkSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            SectionHeaderView(title: "Network", icon: "network")

            VStack(spacing: 10) {
                if let tunnelIP = node.tunnel_ip {
                    propertyRow("Tunnel IP", value: tunnelIP, monospaced: true)
                }
                if let subnet = node.private_subnet {
                    propertyRow("Private Subnet", value: subnet, monospaced: true)
                }
                if let endpoint = node.listen_endpoint {
                    propertyRow("Listen Endpoint", value: endpoint, monospaced: true)
                }
            }
            .cardStyle()
        }
    }

    // MARK: - Keys

    private var keysSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            SectionHeaderView(title: "Cryptographic Keys", icon: "key")

            VStack(spacing: 10) {
                if let mgmtKey = node.mgmt_pubkey {
                    keyRow("Management Key", value: mgmtKey)
                }
                if let wgKey = node.wg_pubkey {
                    keyRow("WireGuard Key", value: wgKey)
                }
            }
            .cardStyle()
        }
    }

    // MARK: - Assignments

    private func assignmentsSection(_ assignments: [NodeAssignment]) -> some View {
        VStack(alignment: .leading, spacing: 12) {
            SectionHeaderView(title: "Assignments (\(assignments.count))", icon: "person.badge.key")

            VStack(spacing: 8) {
                ForEach(assignments) { assignment in
                    VStack(alignment: .leading, spacing: 6) {
                        Text(assignment.management_pubkey)
                            .font(.caption.monospaced())
                            .foregroundColor(.textSecondary)
                            .lineLimit(1)
                            .truncationMode(.middle)

                        HStack(spacing: 4) {
                            ForEach(assignment.permissions, id: \.self) { perm in
                                BadgeView(
                                    text: perm,
                                    color: permissionColor(perm)
                                )
                            }
                        }
                    }
                    .padding(10)
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .background(Color.surfaceDark)
                    .clipShape(RoundedRectangle(cornerRadius: 8))
                }
            }
            .cardStyle()
        }
    }

    // MARK: - Actions

    private var actionsSection: some View {
        VStack(alignment: .leading, spacing: 12) {
            if let message = statusMessage {
                HStack {
                    Image(systemName: "info.circle")
                        .foregroundColor(.blue)
                    Text(message)
                        .font(.caption)
                        .foregroundColor(.textSecondary)
                }
                .padding(10)
                .background(Color.blue.opacity(0.1))
                .clipShape(RoundedRectangle(cornerRadius: 8))
            }

            HStack(spacing: 12) {
                if isEditing {
                    Button(action: { Task { await saveChanges() } }) {
                        HStack {
                            if isSaving {
                                ProgressView()
                                    .scaleEffect(0.7)
                            }
                            Text("Save Changes")
                        }
                    }
                    .buttonStyle(LemonButtonStyle())
                    .disabled(isSaving)

                    Button("Cancel") {
                        isEditing = false
                        editHostname = node.hostname
                        editRegion = node.region ?? ""
                    }
                    .buttonStyle(LemonButtonStyle(isProminent: false))
                }

                Spacer()

                Button(role: .destructive, action: { showDeleteConfirmation = true }) {
                    HStack {
                        Image(systemName: "trash")
                        Text("Delete Node")
                    }
                }
                .buttonStyle(.bordered)
                .tint(.red)
            }
        }
    }

    // MARK: - Helpers

    private var nodeColor: Color {
        switch node.nodeType {
        case .root: return .purple
        case .customer: return .blue
        case .endpoint: return .lemonYellow
        case .relay: return .lemonGreen
        }
    }

    private func propertyRow(_ label: String, value: String, monospaced: Bool = false) -> some View {
        HStack {
            Text(label)
                .font(.subheadline)
                .foregroundColor(.textSecondary)
                .frame(width: 120, alignment: .leading)
            Text(value)
                .font(monospaced ? .subheadline.monospaced() : .subheadline)
                .foregroundColor(.textPrimary)
                .textSelection(.enabled)
            Spacer()
        }
    }

    private func keyRow(_ label: String, value: String) -> some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(label)
                .font(.caption)
                .foregroundColor(.textSecondary)
            HStack {
                Text(value)
                    .font(.caption.monospaced())
                    .foregroundColor(.textPrimary)
                    .lineLimit(2)
                    .textSelection(.enabled)
                Spacer()
                Button(action: {
                    NSPasteboard.general.clearContents()
                    NSPasteboard.general.setString(value, forType: .string)
                    statusMessage = "Copied \(label) to clipboard"
                }) {
                    Image(systemName: "doc.on.doc")
                        .font(.caption)
                        .foregroundColor(.textTertiary)
                }
                .buttonStyle(.plain)
            }
        }
    }

    private func permissionColor(_ perm: String) -> Color {
        switch perm {
        case "read": return .blue
        case "write": return .orange
        case "admin": return .red
        case "manage": return .purple
        default: return .gray
        }
    }

    // MARK: - Actions

    private func saveChanges() async {
        isSaving = true
        statusMessage = nil

        if editHostname != node.hostname {
            let delta = TreeDelta(
                node_id: node.id,
                field: "hostname",
                value: editHostname,
                author_pubkey: appState.publicKeyBase64 ?? "",
                signature: ""
            )
            do {
                let response = try await appState.client.submitTreeDelta(delta)
                if response.success {
                    statusMessage = "Hostname updated"
                    appState.addActivity(.success, "Updated hostname for \(node.id)")
                } else {
                    statusMessage = response.error ?? "Failed to update hostname"
                }
            } catch {
                statusMessage = error.localizedDescription
            }
        }

        isEditing = false
        isSaving = false
    }

    private func deleteNode() async {
        let delta = TreeDelta(
            node_id: node.id,
            field: "delete",
            value: "true",
            author_pubkey: appState.publicKeyBase64 ?? "",
            signature: ""
        )
        do {
            let response = try await appState.client.submitTreeDelta(delta)
            if response.success {
                appState.addActivity(.success, "Deleted node: \(node.hostname)")
            } else {
                appState.addActivity(.error, "Failed to delete: \(response.error ?? "unknown")")
            }
        } catch {
            appState.addActivity(.error, "Delete failed: \(error.localizedDescription)")
        }
    }
}

File: apps/LemonadeNexus/lib/src/views/node_detail_view.dart
/// @title Node Detail View
/// @description Detailed view of a tree node.
///
/// Matches macOS NodeDetailView.swift functionality:
/// - Node header with icon and badges
/// - Properties section
/// - Network info section
/// - Cryptographic keys section
/// - Assignments section
/// - Delete node action

import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../state/providers.dart';
import '../state/app_state.dart';
import '../sdk/models.dart';
import '../../theme/app_theme.dart';
import '../../theme/components.dart';
import 'tree_browser_view.dart' show NodeType;

class NodeDetailView extends ConsumerStatefulWidget {
  final TreeNode node;

  const NodeDetailView({super.key, required this.node});

  @override
  ConsumerState<NodeDetailView> createState() => _NodeDetailViewState();
}

class _NodeDetailViewState extends ConsumerState<NodeDetailView> {
  bool _isEditing = false;
  bool _isSaving = false;
  String? _statusMessage;

  @override
  Widget build(BuildContext context) {
    final node = widget.node;
    final nodeType = NodeType.fromRaw(node.nodeType);
    final scheme = Theme.of(context).colorScheme;

    return SingleChildScrollView(
      padding: const EdgeInsets.all(24),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          // Header
          _buildHeader(node, nodeType),
          Divider(color: scheme.outline, height: 24),
          // Properties
          _buildPropertiesSection(node),
          Divider(color: scheme.outline, height: 24),
          // Network
          _buildNetworkSection(node, nodeType),
          Divider(color: scheme.outline, height: 24),
          // Keys
          _buildKeysSection(node),
          // Assignments
          if (node.assignments != null && node.assignments!.isNotEmpty) ...[
            Divider(color: scheme.outline, height: 24),
            _buildAssignmentsSection(node.assignments!),
          ],
          const SizedBox(height: 20),
          // Actions
          _buildActionsSection(node),
        ],
      ),
    );
  }

  Widget _buildHeader(TreeNode node, NodeType nodeType) {
    return Row(
      children: [
        // Icon
        Container(
          width: 56,
          height: 56,
          decoration: BoxDecoration(
            color: _nodeColor(nodeType).withValues(alpha: 0.15),
            borderRadius: BorderRadius.circular(12),
          ),
          child: Icon(
            nodeType.icon,
            color: _nodeColor(nodeType),
            size: 28,
          ),
        ),
        const SizedBox(width: 16),
        // Title and badges
        Expanded(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(
                node.displayName,
                style: const TextStyle(
                  fontSize: 18,
                  fontWeight: FontWeight.bold,
                ),
              ),
              const SizedBox(height: 4),
              Row(
                children: [
                  LemonBadge(text: nodeType.displayName, color: _nodeColor(nodeType)),
                  const SizedBox(width: 8),
                  _buildIdBadge(node.id),
                ],
              ),
            ],
          ),
        ),
        // Edit button
        IconButton(
          icon: Icon(
            _isEditing ? Icons.check_circle : Icons.edit,
            color: _isEditing
                ? AppTheme.lemonGreen
                : Theme.of(context).colorScheme.onSurfaceVariant,
            size: 24,
          ),
          onPressed: () => setState(() => _isEditing = !_isEditing),
          tooltip: _isEditing ? 'Done Editing' : 'Edit Node',
        ),
      ],
    );
  }

  Widget _buildPropertiesSection(TreeNode node) {
    return _buildSection(
      icon: Icons.info_outline,
      title: 'Properties',
      child: Column(
        children: [
          _buildPropertyRow('Node ID', value: node.id, monospaced: true),
          _buildPropertyRow('Parent ID', value: node.parentId, monospaced: true),
          _buildPropertyRow('Type', value: NodeType.fromRaw(node.nodeType).displayName),
          _buildPropertyRow('Hostname', value: node.displayName),
          if (node.displayRegion != null)
            _buildPropertyRow('Region', value: node.displayRegion!),
        ],
      ),
    );
  }

  Widget _buildNetworkSection(TreeNode node, NodeType nodeType) {
    if (nodeType == NodeType.customer || nodeType == NodeType.root) {
      final scheme = Theme.of(context).colorScheme;
      return _buildSection(
        icon: Icons.lan,
        title: 'Network',
        child: Row(
          children: [
            Icon(Icons.info, size: 16, color: scheme.onSurfaceVariant),
            const SizedBox(width: 8),
            Expanded(
              child: Text(
                nodeType == NodeType.root
                    ? 'Root node manages the network but does not have a tunnel address.'
                    : 'Group nodes organize endpoints. Select a child endpoint to see network details.',
                style: TextStyle(
                  color: scheme.onSurfaceVariant,
                  fontSize: 12,
                ),
              ),
            ),
          ],
        ),
      );
    }

    return _buildSection(
      icon: Icons.lan,
      title: 'Network',
      child: Column(
        children: [
          if (node.displayTunnelIp != null)
            _buildPropertyRow('Tunnel IP', value: node.displayTunnelIp!, monospaced: true),
          if (node.privateSubnet != null)
            _buildPropertyRow('Private Subnet', value: node.privateSubnet!, monospaced: true),
          if (node.listenEndpoint != null)
            _buildPropertyRow('Listen Endpoint', value: node.listenEndpoint!, monospaced: true),
          if (node.displayTunnelIp == null &&
              node.privateSubnet == null &&
              node.listenEndpoint == null)
            Text(
              'No network info assigned yet.',
              style: TextStyle(
                color: Theme.of(context).colorScheme.onSurfaceVariant,
                fontSize: 12,
              ),
            ),
        ],
      ),
    );
  }

  Widget _buildKeysSection(TreeNode node) {
    return _buildSection(
      icon: Icons.vpn_key,
      title: 'Cryptographic Keys',
      child: Column(
        children: [
          if (node.mgmtPubkey != null)
            _buildKeyRow('Management Key', value: node.mgmtPubkey!),
          if (node.wgPubkey != null)
            _buildKeyRow('WireGuard Key', value: node.wgPubkey!),
          if (node.mgmtPubkey == null && node.wgPubkey == null)
            Text(
              'No keys available.',
              style: TextStyle(
                color: Theme.of(context).colorScheme.onSurfaceVariant,
                fontSize: 12,
              ),
            ),
        ],
      ),
    );
  }

  Widget _buildAssignmentsSection(List<NodeAssignment> assignments) {
    return _buildSection(
      icon: Icons.badge,
      title: 'Assignments (${assignments.length})',
      child: Column(
        children: assignments.map(_buildAssignmentCard).toList(),
      ),
    );
  }

  Widget _buildAssignmentCard(NodeAssignment assignment) {
    final scheme = Theme.of(context).colorScheme;
    return Padding(
      padding: const EdgeInsets.only(bottom: 8),
      child: Container(
        padding: const EdgeInsets.all(10),
        decoration: BoxDecoration(
          color: scheme.surfaceContainerHighest,
          borderRadius: BorderRadius.circular(8),
        ),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              assignment.managementPubkey,
              style: TextStyle(
                color: scheme.onSurfaceVariant,
                fontSize: 11,
                fontFamily: 'monospace',
              ),
              maxLines: 1,
              overflow: TextOverflow.ellipsis,
            ),
            const SizedBox(height: 6),
            Wrap(
              spacing: 4,
              runSpacing: 4,
              children: assignment.permissions
                  .map((perm) => LemonBadge(
                        text: perm,
                        color: _permissionColor(perm),
                      ))
                  .toList(),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildActionsSection(TreeNode node) {
    final scheme = Theme.of(context).colorScheme;
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        if (_statusMessage != null) ...[
          Container(
            padding: const EdgeInsets.all(10),
            decoration: BoxDecoration(
              color: AppTheme.infoColor.withValues(alpha: 0.1),
              borderRadius: BorderRadius.circular(8),
            ),
            child: Row(
              children: [
                const Icon(Icons.info, color: AppTheme.infoColor, size: 16),
                const SizedBox(width: 8),
                Expanded(
                  child: Text(
                    _statusMessage!,
                    style: TextStyle(color: scheme.onSurfaceVariant, fontSize: 12),
                  ),
                ),
              ],
            ),
          ),
          const SizedBox(height: 12),
        ],
        Row(
          children: [
            if (_isEditing) ...[
              Expanded(
                child: ElevatedButton.icon(
                  onPressed: _isSaving ? null : () => _saveChanges(node),
                  icon: _isSaving
                      ? const SizedBox(
                          width: 14,
                          height: 14,
                          child: CircularProgressIndicator(strokeWidth: 2),
                        )
                      : const Icon(Icons.save),
                  label: const Text('Save Changes'),
                  style: ElevatedButton.styleFrom(
                    backgroundColor: AppTheme.lemonYellowDark,
                    foregroundColor: Colors.black,
                    padding: const EdgeInsets.symmetric(vertical: 12),
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
                  ),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: ElevatedButton(
                  onPressed: () => setState(() => _isEditing = false),
                  style: ElevatedButton.styleFrom(
                    backgroundColor: scheme.surfaceContainerHighest,
                    foregroundColor: scheme.onSurface,
                    padding: const EdgeInsets.symmetric(vertical: 12),
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
                  ),
                  child: const Text('Cancel'),
                ),
              ),
              const SizedBox(width: 12),
            ],
            Expanded(
              child: ElevatedButton.icon(
                onPressed: () => _showDeleteConfirmationDialog(node),
                icon: const Icon(Icons.delete),
                label: const Text('Delete Node'),
                style: ElevatedButton.styleFrom(
                  backgroundColor: AppTheme.errorColor,
                  foregroundColor: Colors.white,
                  padding: const EdgeInsets.symmetric(vertical: 12),
                  shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
                ),
              ),
            ),
          ],
        ),
      ],
    );
  }

  Widget _buildSection({required IconData icon, required String title, required Widget child}) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        SectionHeader(title: title, icon: icon),
        const SizedBox(height: 12),
        AppCard(child: child),
      ],
    );
  }

  Widget _buildPropertyRow(String label, {required String value, bool monospaced = false}) {
    final scheme = Theme.of(context).colorScheme;
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 6),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          SizedBox(
            width: 120,
            child: Text(
              label,
              style: TextStyle(color: scheme.onSurfaceVariant, fontSize: 12),
            ),
          ),
          Expanded(
            child: Text(
              value,
              style: TextStyle(
                fontSize: 12,
                fontFamily: monospaced ? 'monospace' : null,
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildKeyRow(String label, {required String value}) {
    final scheme = Theme.of(context).colorScheme;
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            label,
            style: TextStyle(color: scheme.onSurfaceVariant, fontSize: 11),
          ),
          const SizedBox(height: 4),
          Row(
            children: [
              Expanded(
                child: Text(
                  value,
                  style: const TextStyle(
                    fontSize: 11,
                    fontFamily: 'monospace',
                  ),
                  maxLines: 2,
                  overflow: TextOverflow.ellipsis,
                ),
              ),
              const SizedBox(width: 8),
              IconButton(
                icon: const Icon(Icons.copy, size: 16),
                color: scheme.onSurfaceVariant,
                onPressed: () async {
                  await Clipboard.setData(ClipboardData(text: value));
                  if (!mounted) return;
                  setState(() => _statusMessage = 'Copied $label to clipboard');
                  Future.delayed(const Duration(seconds: 3), () {
                    if (mounted) setState(() => _statusMessage = null);
                  });
                },
                tooltip: 'Copy to clipboard',
                padding: EdgeInsets.zero,
                constraints: const BoxConstraints(),
              ),
            ],
          ),
        ],
      ),
    );
  }

  Widget _buildIdBadge(String id) {
    final scheme = Theme.of(context).colorScheme;
    final shortId = id.length > 12
        ? '${id.substring(0, 6)}...${id.substring(id.length - 4)}'
        : id;
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 3),
      decoration: BoxDecoration(
        color: scheme.surfaceContainerHighest,
        borderRadius: BorderRadius.circular(4),
      ),
      child: Text(
        shortId,
        style: TextStyle(
          color: scheme.onSurfaceVariant,
          fontSize: 9,
          fontFamily: 'monospace',
        ),
      ),
    );
  }

  Color _nodeColor(NodeType type) {
    switch (type) {
      case NodeType.root:
        return const Color(0xFF9B5DE5);
      case NodeType.customer:
        return AppTheme.infoColor;
      case NodeType.endpoint:
        return AppTheme.lemonYellowDark;
      case NodeType.relay:
        return AppTheme.lemonGreen;
    }
  }

  Color _permissionColor(String perm) {
    switch (perm) {
      case 'read':
        return AppTheme.infoColor;
      case 'write':
        return AppTheme.nodeOrange;
      case 'admin':
        return AppTheme.errorColor;
      case 'manage':
        return const Color(0xFF9B5DE5);
      default:
        return Colors.grey;
    }
  }

  Future<void> _saveChanges(TreeNode node) async {
    setState(() {
      _isSaving = true;
      _statusMessage = null;
    });

    // TODO: Implement actual save functionality when edit fields are added.
    // For now this is a placeholder; AppNotifier currently exposes no
    // "update node properties" entry point.
    await Future<void>.delayed(const Duration(milliseconds: 200));

    if (!mounted) return;
    setState(() {
      _isEditing = false;
      _isSaving = false;
      _statusMessage = 'Changes saved (not yet wired to SDK)';
    });

    Future.delayed(const Duration(seconds: 3), () {
      if (mounted) setState(() => _statusMessage = null);
    });
  }

  void _showDeleteConfirmationDialog(TreeNode node) {
    final scheme = Theme.of(context).colorScheme;
    showDialog<void>(
      context: context,
      builder: (dialogContext) => AlertDialog(
        backgroundColor: scheme.surface,
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(12),
          side: BorderSide(color: scheme.outline),
        ),
        title: const Text('Delete Node'),
        content: Text(
          'Are you sure you want to delete "${node.displayName}"? This action cannot be undone.',
          style: TextStyle(color: scheme.onSurfaceVariant, fontSize: 13),
        ),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(dialogContext),
            child: const Text('Cancel'),
          ),
          ElevatedButton(
            onPressed: () {
              Navigator.pop(dialogContext);
              _deleteNode(node);
            },
            style: ElevatedButton.styleFrom(
              backgroundColor: AppTheme.errorColor,
              foregroundColor: Colors.white,
            ),
            child: const Text('Delete'),
          ),
        ],
      ),
    );
  }

  Future<void> _deleteNode(TreeNode node) async {
    final notifier = ref.read(appNotifierProvider.notifier);
    try {
      final success = await notifier.deleteNode(nodeId: node.id);
      if (!mounted) return;
      if (success) {
        notifier.addActivity(
          ActivityLevel.success,
          'Deleted node: ${node.displayName}',
        );
        Navigator.pop(context); // Pop back to tree browser
      } else {
        notifier.addActivity(
          ActivityLevel.error,
          'Failed to delete: ${node.displayName}',
        );
      }
    } catch (e) {
      notifier.addActivity(ActivityLevel.error, 'Delete failed: $e');
    }
  }
}
